integration/secret/secret_test.go

```diff
@@ -41,3 +43,110 @@ func TestSecretInspect(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, secret.ID, secretResp.ID)
 }
+
+func TestSecretList(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux")
+
+	defer setupTest(t)()
+	d := swarm.NewSwarm(t, testEnv)
+	defer d.Stop(t)
+	client, err := request.NewClientForHost(d.Sock())
+	require.NoError(t, err)
+
+	ctx := context.Background()
+
+	testName0 := "test0"
+	testName1 := "test1"
+
+	// create secret test0
+	secret0, err := client.SecretCreate(ctx, swarmtypes.SecretSpec{
+		Annotations: swarmtypes.Annotations{
+			Name:   testName0,
+			Labels: map[string]string{"type": "test"},
+		},
+		Data: []byte("TESTINGDATA0"),
+	})
+	require.NoError(t, err)
+	assert.NotEqual(t, secret0.ID, "")
+
+	secret, _, err := client.SecretInspectWithRaw(ctx, secret0.ID)
+	require.NoError(t, err)
+	assert.Equal(t, secret.Spec.Name, testName0)
```

Comment:
Is this inspect and checking the name necessary? This is supposed to be testing list not inspect, right? I think we can remove the extra inspect.